config/http_config.go

```diff
@@ -424,12 +424,14 @@ func NewRoundTripperFromConfig(cfg HTTPClientConfig, name string, optFuncs ...HT
 		return nil, err
 	}
 
-	if len(cfg.TLSConfig.CAFile) == 0 {
+	if len(cfg.TLSConfig.getCAName()) == 0 {
```

Comment:
All the changes like this are meant to reduce the noise around having to check whether the filename is set or the inline version is set.